agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -1230,13 +1230,90 @@ public void validate() {
     }
   }
 
+  public enum SpanFilterType {
+    REGEX
+  }
+
+  public static class SpanFilter {
+    public SpanFilterType type;
+    public String value;
+  }
+
+  public static class SpanAggregationConfig {
+
+    // Threshold in ms over which a span will consider to be a breach
+    // Used by the breach ratio aggregation
+    public int thresholdMs = 5000;
+
+    // Minimum number of samples that must have been collected in order for the aggregation to
+    // produce data. Avoids volatile aggregation output on small sample sizes.
+    public int minimumSamples = 0;
+  }
+
+  public enum SpanAggregationType {
+    BREACH_RATIO
+  }
+
+  public static class SpanAggregation {
+    public SpanAggregationType type = SpanAggregationType.BREACH_RATIO;
+    public long windowSize = 60000; // in ms
+    public String evaluationFrequency = "every_sample";
+    public SpanAggregationConfig configuration = new SpanAggregationConfig();
+  }
+
+  public enum SpanTriggerThresholdType {
+    GREATER_THAN
+  }
+
+  public static class SpanTriggerThreshold {
+    public SpanTriggerThresholdType type = SpanTriggerThresholdType.GREATER_THAN;
+
+    // Threshold value applied to the output of the aggregation
+    // i.e :
+    //  - For the BreachRatio aggregation, 0.75 means this will trigger if 75% of sample breach the
+    // threshold.
+    //  - For a rolling average aggregation 0.75 will mean this will trigger if the average span
+    // processing time
+    //      breaches 0.75ms
+    public float value = 0.75f;
```

Comment:
just confirming my understanding: this means the default trigger is based on 75th percentile exceeding 5000 milliseconds?

Reply:
correct